js/data.js

```diff
@@ -6,3 +6,19 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+var previousData = localStorage.getItem('code-data-storage');
+var previousEntries = localStorage.getItem('code-entry-storage');
+if (previousData !== null) {
+  data = JSON.parse(previousData);
+}
+if (previousEntries !== null) {
+  data.entries = JSON.parse(previousEntries);
+}
+
+window.addEventListener('beforeunload', function (event) {
+  event.preventDefault();
+  var entriesJSON = JSON.stringify(data.entries);
+  var dataJSON = JSON.stringify(data);
+  this.localStorage.setItem('code-data-storage', dataJSON);
```

Comment:
Instead of storing the entries separately, I would just save the whole object and replace it when the app starts with the stored version. also, you don't need the `this` keyword for localStorage.